src/arithmetic/algebraic_expression/algebraic_expression_optimization.c

```diff
@@ -328,17 +339,19 @@ static void _Pushdown_TransposeExp
 		_Pushdown_TransposeOperation(exp);
 		break;
 	case AL_OPERAND:
-		_Pushdown_TransposeOperand(exp);
+		_AlgebraicExpression_TransposeOperand(exp);
 		break;
 	default:
 		assert("unknown algebraic expression node type" && false);
 		break;
 	}
 }
 
-/* Push down transpose operations to the point where they are applied to individual operands
- * once this optimization is applied there shouldn't be instances of transpose acting on
- * operation nodes such as multiplication and addition.
+/* Push down intermediate transpose operations to the point where they are applied to
+ * individual operands, rather than operation nodes such as multiplication and addition.
+ * Then transpose the operands and replace the operation nodes with them.
+ * Once this optimization is applied, there shouldn't be any transpose operation nodes
+ * in the tree, and individual operands will be transposed matrices where appropriate.
```

Comment:
Please break this into two different optimisations, one `_AlgebraicExpression_PushDownTranspose` which simply relocates nodes

And another optimisations function which looks for transpose nodes applied directly to operands and performs the transpose, replacing the transpose operation node and the operand node with just a single transposed operand node.